File: src/main/java/com/dh/ondot/member/domain/service/WithdrawalService.java
package com.dh.ondot.member.domain.service;

import com.dh.ondot.member.domain.MemberWithdrawal;
import com.dh.ondot.member.domain.repository.MemberWithdrawalRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class WithdrawalService {
    private final MemberWithdrawalRepository withdrawalRepository;

    public void withdrawMember(Long memberId, Long withdrawalReasonId, String customReason) {
        MemberWithdrawal withdrawal = MemberWithdrawal.create(
                memberId,
                withdrawalReasonId,
                customReason
        );
        withdrawalRepository.save(withdrawal);
    }
}

File: src/main/kotlin/com/dh/ondot/member/domain/service/WithdrawalService.kt
package com.dh.ondot.member.domain.service

import com.dh.ondot.member.domain.MemberWithdrawal
import com.dh.ondot.member.domain.repository.MemberWithdrawalRepository
import org.springframework.stereotype.Service

@Service
class WithdrawalService(
    private val withdrawalRepository: MemberWithdrawalRepository,
) {
    fun saveWithdrawalReason(memberId: Long, withdrawalReasonId: Long, customReason: String?) {
        val withdrawal = MemberWithdrawal.create(
            memberId,
            withdrawalReasonId,
            customReason,
        )
        withdrawalRepository.save(withdrawal)
    }
}
